src/CommunityToolkit.Maui/Services/DialogFragmentService.android.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Android.Content;
+using Android.OS;
+using Android.Views;
+using AndroidX.AppCompat.App;
+using CommunityToolkit.Maui.Core;
+using Debug = System.Diagnostics.Debug;
+using DialogFragment = AndroidX.Fragment.App.DialogFragment;
+using Fragment = AndroidX.Fragment.App.Fragment;
+using FragmentManager = AndroidX.Fragment.App.FragmentManager;
+
+namespace CommunityToolkit.Maui.Services;
+
+sealed class DialogFragmentService : IDialogFragmentService
+{
+	public void OnFragmentAttached(FragmentManager fm, Fragment f, Context context)
+	{
+	}
+
+	public void OnFragmentCreated(FragmentManager fm, Fragment f, Bundle? savedInstanceState)
+	{
+	}
+
+	public void OnFragmentDestroyed(FragmentManager fm, Fragment f)
+	{
+	}
+
+	public void OnFragmentDetached(FragmentManager fm, Fragment f)
+	{
+	}
+
+	public void OnFragmentPaused(FragmentManager fm, Fragment f)
+	{
+	}
+
+	public void OnFragmentPreAttached(FragmentManager fm, Fragment f, Context context)
+	{
+	}
+
+	public void OnFragmentPreCreated(FragmentManager fm, Fragment f, Bundle? savedInstanceState)
+	{
+	}
+
+	public void OnFragmentResumed(FragmentManager fm, Fragment f)
+	{
+	}
+
+	public void OnFragmentSaveInstanceState(FragmentManager fm, Fragment f, Bundle outState)
+	{
+	}
+
+	public void OnFragmentStarted(FragmentManager fm, Fragment f)
+	{
+		if (!IsDialogFragment(f, out var dialogFragment) || Platform.CurrentActivity is not AppCompatActivity activity)
+		{
```

Comment:
If this fails do we want to output a message to developer?

Reply:
No, if it fails is because the Fragment isn't a DialogFragment and we want to silently ignore it. The CurrentActivity check is just to make sure isn't null, I used the strongly typed to reuse it, and we already check and log when the Activity isn't the desired type